### PR TITLE
safestrlib: only declare the safestrlib functions if needed

### PR DESCRIPTION
**Pull request by 1480c1**
# Description

didn't realize I was missing some code

The rest is for making sure that the unused functions are not declared at all so they do not conflict with the regular functions that already exist

# Issue

# Author(s)
@1480c1

# Performance impact

- [ ] quality
- [ ] memory
- [ ] speed
- [ ] 8 bit
- [ ] 10 bit
- [x] N/A

# Test set
- [ ] obj-1-fast can be found [here](https://media.xiph.org/video/derf/objective-1-fast.tar.gz)
- [ ] other
- [x] N/A

# Merge method
- [x] Allow the maintainer to squash and merge when PR is ready to create a 1-commit to the master branch. The maintainer will be able to fix typos / combine commit messages to create a more readable 1-commit message or use whatever is stated in the 'Description' section
- [ ] I will clean up my commits and the maintainer shall use 'rebase and merge' to the master branch

### CODE REVIEW

**Comment by 1480c1** [comment]
seems there are some other issues with this one due to conflicting types with the mingw-w64 gcc compiler on ubuntu, ignore for now

File: third_party/safestringlib/safe_str_lib.h [hunk 1]
```diff
@@ -71,8 +73,14 @@ extern errno_t strcmp_s(const char *dest, rsize_t dmax, const char *src, int *in
...
 extern errno_t strcmpfld_s(const char *dest, rsize_t dmax, const char *src, int *indicator);
 
+#endif
+
+#ifndef HAVE_STRCPY_S
 /* string copy */
 extern errno_t strcpy_s(char *dest, rsize_t dmax, const char *src);
+#endif
+
+#if 0
```
Comment (hassount) [inline third_party/safestringlib/safe_str_lib.h:83]: is the choice here between #if 0 and removing the code ?

Comment (1480c1) [reply third_party/safestringlib/safe_str_lib.h:83]: I used an `#if 0` so it's easier to see if there are any changes if safestringlib updates it in the future as using this seems to be the least destructive way to do this

**Comment by paulsgh** [comment]
Tested a mix of clips comprised of different resolutions and bit depths.
This PR is lossless.

| | Speed Deviation  | BD-Rate   Deviation | Memory Deviation
-- | -- | -- | --
M0 - 8bit - 2pass| 0.01% | 0.00% | 0.01%
M0 - 10bit - 2pass| 0.02% | 0.00% | -0.04%
M0 - 8bit - 1pass| -0.02% | 0.00% | -0.02%
M4 - 8bit - 1pass| 0.09% | 0.00% | 0.02%
M8 - 8bit - 1pass| 0.00% | 0.00% | -0.02%
M0 - 10bit - 1pass| 0.02% | 0.00% | 0.05%
M4 - 10bit - 1pass| 0.20% | 0.00% | -0.02%
M8 - 10bit - 1pass| 0.23% | 0.00% | -0.01%

**Pull request merged**